Framework/PythonInterface/test/python/plugins/algorithms/WorkflowAlgorithms/D7YIGPositionCalibrationTest.py

```diff
@@ -0,0 +1,44 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+import unittest
+from mantid.simpleapi import config, mtd, D7YIGPositionCalibration, Load, LoadILLPolarizedDiffraction
+from mantid.api import ITableWorkspace
+import os.path
+from os import path
+
+
+class D7YIGPositionCalibrationTest(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        config.appendDataSearchSubDir('ILL/D7/')
+
+    def setUp(self):
+        Load('402652_403041.nxs', OutputWorkspace='shortWavelengthScan')
+
+    def tearDown(self):
+        mtd.clear()
+        if path.exists('test_shortWavelength.xml'):
+            os.remove('test_shortWavelength.xml')
+
+    def test_algorithm_with_no_input_workspace_raises_exception(self):
+        with self.assertRaises(RuntimeError):
+            D7YIGPositionCalibration()
+
+    def test_shortWavelength(self):
+        approximate_wavelength = '3.14' # Angstrom
+        self.assertTrue(mtd['shortWavelengthScan'])
+        D7YIGPositionCalibration(InputWorkspace='shortWavelengthScan', ApproximateWavelength=approximate_wavelength,
+                                 YIGPeaksFile='D7_YIG_peaks.xml', CalibrationOutputFile='test_shortWavelength.xml',
```

Comment:
For the output file, make sure it is in `/tmp` directory, which you should get from `os` since it is different on different platforms.

Reply:
The output is saved by default in the user-defined Mantid Default Save Directory. This can be either made more explicit where to delete the file or the saving path needs to be changed in the algorithm.